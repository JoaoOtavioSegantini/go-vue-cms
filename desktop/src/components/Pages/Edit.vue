<template>
  <div class="container-fluid">
    <div class="row">
      <div class="col-1 nav">
        <p class="display-6">
          <small>AÇÕES</small>
        </p>

        <a :href="'/pages/' + page.ID" class="btn btn-secondary" title="Voltar">
          <i class="fa fa-arrow-circle-left"></i>
        </a>
      </div>
      <div class="col-11">
        <div class="card">
          <div class="card-body">
            <h5 class="title">
              <i class="fa fa-file-text-o"></i> {{ page.title }}
              <small>Edição de página do site</small>
            </h5>

            <form @submit.prevent="save()">
              <div class="form-group">
                <label for="title">Título</label>
                <input
                  type="text"
                  class="form-control"
                  id="title"
                  placeholder="Título da página"
                  v-model="page.title"
                />
              </div>
              <div class="form-group">
                <label for="slug">URL</label>
                <input
                  type="text"
                  class="form-control"
                  id="slug"
                  placeholder="Url da página"
                  v-model="page.slug"
                />
              </div>
              <div class="form-group">
                <label for="body">Conteúdo</label>
                <ckeditor :editor="editor" v-model="page.body"></ckeditor>
              </div>
              <button type="submit" class="btn btn-primary">Salvar</button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
export default {
  data() {
    return {
      editor: ClassicEditor,
    };
  },
  computed: {
    page() {
      return { ...this.$store.state.Pages.onePage };
    },
  },
  methods: {
    save() {
      this.$store
        .dispatch("updatePage", this.page)
        // eslint-disable-next-line no-unused-vars
        .then((_res) => {
          this.$router.push({ path: "/pages/" + this.page.ID });
        });
    },
  },
  mounted() {
    this.$store
      .dispatch("getPage", this.$route.params.id)
      // eslint-disable-next-line no-unused-vars
      .then((_res) => {
        console.log(this.page.body);
      })
      .catch((err) => {
        console.error(err.stack);
      });
  },
};
</script>

<style>
.ck-editor__editable {
  min-height: 260px;
}
</style>
