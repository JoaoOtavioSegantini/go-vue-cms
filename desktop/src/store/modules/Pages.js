const state = {
  all: [],
  onePage: {},
};

const mutations = {
  updatePagesList(state, res) {
    state.all = res.data;
  },
  updatePage(state, res) {
    state.onePage = res.data;
  },
  addToPagesList(state, res) {
    state.all.push(res.data);
  },
};

const actions = {
  listPages(context) {
    window.axios.get("/api/v1/res-data").then((res) => {
      context.commit("updatePagesList", res);
    });
  },
  getPage(context, id) {
    return window.axios.get("/api/v1/res-data/" + id).then((res) => {
      context.commit("updatePage", res);
    });
  },
  createPage(context, data) {
    // let qs = require("qs");

    let config = {
      swalTitle: "Página salva com sucesso",
      swalMessage: "Sua nova página já está disponível",
    };

    return window.axios.post("/api/v1/res-data", data, config).then((res) => {
      context.commit("addToPagesList", res);
      return res;
    });
  },
  updatePage(context, data) {
    let id = data.ID;

    let config = {
      swalTitle: "Página salva com sucesso",
      swalMessage: "Sua nova página já está disponível",
    };

    return window.axios
      .put("/api/v1/res-data/" + id, data, config)
      .then((res) => {
        return res;
      });
  },
  removePage(context, id) {
    return window.axios.delete("/api/v1/res-data/" + id).then((res) => {
      return res;
    });
  },
};

export default {
  state,
  mutations,
  actions,
};
