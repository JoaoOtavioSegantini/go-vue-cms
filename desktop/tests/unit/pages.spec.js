import Pages from "@/components/Pages.vue";
import { mount } from "@vue/test-utils";
import { createStore } from "vuex";

describe("Pages.vue", () => {
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Pages: {
            all: [{ ID: "1", title: "Post 1", body: "A simple page text body" }],
          },
        };
      },
      mutations: {
        updatePagesList(state) {
          state.Pages = [];
        },
      },
      actions: {
        listPages(ctx) {
          const res = mockGet();
          ctx.commit("updatePagesList", res);
        },
      },
    });
  };

  function factory() {
    const store = createVuexStore();
    return mount(Pages, {
      global: {
        plugins: [store],
      },
    });
  }

  const mockGet = jest.fn();

  jest.mock("axios", () => ({
    get: () => mockGet(),
  }));

  it("renders props when passed", () => {
    const msg = "AÇÕES Páginas Gerenciamento de páginas do site#titlebody1Post 1A simple page text bodyver";

    const wrapper = factory();

    expect(mockGet).toHaveBeenCalled();
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
