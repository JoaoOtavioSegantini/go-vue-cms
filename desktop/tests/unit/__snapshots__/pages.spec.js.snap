// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages.vue renders props when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/pages/add\\" class=\\"btn btn-secondary\\" title=\\"Nova página\\"><i class=\\"fa fa-plus\\"></i></a>
    </div>
    <div class=\\"col\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Páginas <small>Gerenciamento de páginas do site</small></h5>
          <table class=\\"table\\" aria-label=\\"tabela contendo uma lista de páginas\\">
            <thead>
              <tr>
                <th scope=\\"col\\">#</th>
                <th scope=\\"col\\">title</th>
                <th scope=\\"col\\">body</th>
                <th scope=\\"col\\"></th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope=\\"row\\">1</th>
                <td>Post 1</td>
                <td>
                  <div class=\\"cut-text\\">A simple page text body</div>
                </td>
                <td class=\\"text-right\\"><a href=\\"/pages/1\\" class=\\"btn btn-primary btn-sm\\">ver</a></td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
